src/ssl.c

```diff
@@ -14604,19 +14634,21 @@ int AddSession(WOLFSSL* ssl)
     session->bornOn  = LowResTimer();
 
 #ifdef HAVE_SESSION_TICKET
-    /* Check if another thread modified ticket since alloc */
-    if ((word16)ticLen != ssl->session.ticketLen) {
-        error = VAR_STATE_CHANGE_E;
+    /* Cleanup cache row's old Dynamic buff if exists */
+    if (session->isDynamic) {
+        XFREE(session->ticket, ssl->heap, DYNAMIC_TYPE_SESSION_TICK);
+        session->ticket = NULL;
     }
 
-    if (error == 0) {
-        /* Cleanup cache row's old Dynamic buff if exists */
-        if (session->isDynamic) {
-            XFREE(session->ticket, ssl->heap, DYNAMIC_TYPE_SESSION_TICK);
-            session->ticket = NULL;
+    /* If too large to store in static buffer, use dyn buffer */
+    ticLen = ssl->session.ticketLen;
+    if (ticLen > SESSION_TICKET_LEN) {
+        tmpBuff = (byte*)XMALLOC(ticLen, ssl->heap, DYNAMIC_TYPE_SESSION_TICK);
```

Comment:
Memory allocation can be expensive.
Prefer to do outside lock.

Reply:
Fixed.